roles/ceph-config/tasks/main.yml

```diff
@@ -27,16 +27,50 @@
         - lvm_volumes | default([]) | length > 0
         - osd_scenario == 'lvm'
 
-    # This is a best guess. Ideally we'd like to use `ceph-volume lvm batch --report` to get
-    # a more accurate number but the ceph.conf needs to be in place before that is possible.
-    # There is a tracker to add functionality to ceph-volume which would allow doing this
-    # without the need for a ceph.conf: http://tracker.ceph.com/issues/36088
-    - name: count number of osds for lvm batch scenario
+    - name: run 'ceph-volume lvm batch --report' to see how many osds are to be created
+      ceph_volume:
+        cluster: "{{ cluster }}"
+        objectstore: "{{ osd_objectstore }}"
+        batch_devices: "{{ devices }}"
+        osds_per_device: "{{ osds_per_device | default(1) | int }}"
+        journal_size: "{{ journal_size }}"
+        block_db_size: "{{ block_db_size }}"
+        report: true
+        action: "batch"
+      register: lvm_batch_report
+      environment:
+        CEPH_VOLUME_DEBUG: 1
+      when:
+        - devices | default([]) | length > 0
+        - osd_scenario == 'lvm'
```

Comment:
shouldn't we run this on an osd node?

Reply:
Yeah, all these tasks are in a block that only run on osd nodes. https://github.com/ceph/ceph-ansible/pull/3190/files/497b905dd0acdf6a98d90e8953c57145b4a4732d#diff-408c9554f34e961d163f68b645ee6da3R75

Reply:
Oops, missed that.